Clarify how ranking Qualifications are selected. (#6)
thewca/wcif: specification.md

| Attribute | Type | Description |
| --- | --- | --- |
| `whenDate` | [`Date`](#date) | The date by which the qualification requirement must be satisfied.  If a result is set in a multiple-day competition which begins by this date, that is considered to have been set by this date. |

> | `type` | `"attemptResult"\|"ranking"\|"anyResult"` | The type of Qualification standard. |

Reviewer: ```suggestion
| `type` | `"attemptResult"\|"ranking"\|"anyResult"` | The type of qualification. Either of ranking (top N competitors), attempt result (competitors with result better than Y - either single or average) or any result (competitors with any successful result of the given type). |
```

| Attribute | Type | Description |
| --- | --- | --- |
| `whenDate` | [`Date`](#date) | The date by which the qualification requirement must be satisfied.  If a result is set in a multiple-day competition which begins by this date, that is considered to have been set by this date. |
| `type` | `"single"\|"average"` | The type of result the requirement refers to. |

> | `method` | `"result"\|"ranking"\|"any_result"` | The method for determining which competitors qualify. |

Reviewer: To stay consistent with `AdvancementCondition` let's use `"attemptResult"` instead of `"result"`. Also, the spec generally uses camel case, so `any_result -> anyResult`.

Looking at `AdvancementCondition`, I'm also wondering: `method -> type`, `type -> resultType/variant`, thoughts?

> | `level` | [`AttemptResult`](#attemptresult)\|[`Ranking`](#ranking)\|null | The parameter of the qualification condition of the given type. Not used with `any_result`. |

Reviewer: ```suggestion
| `level` | [`AttemptResult`](#attemptresult)\|[`Ranking`](#ranking)\|`null` | The parameter of the qualification condition of the given type. Not used with `any_result`. |
```

#### Examples

```json
{
  "whenDate": "2020-04-25",
  "type": "single",
  "method": "result",
  "level": 6000
}
```

```json
{
  "whenDate": "2020-04-25",
  "type": "average",
  "method": "ranking",
  "level": 50
}
```

```json
{
  "whenDate": "2020-04-25",
  "type": "single",

>   "method: "any_result",

Reviewer 2: Missed a quotation mark

Author: Ah, thank you! Fixed.

{
  "whenDate": "2020-04-25",
  "type": "ranking",
  "resultType": "average",
  "level": 50
}
```

```json
{
  "whenDate": "2020-04-25",
  "type": "anyResult",

>   "resultType": "single",

Reviewer: ```suggestion
  "resultType": "single",
  "level": null
```
